src/definitions/elements/input.less

```diff
@@ -455,20 +455,34 @@
   margin: 0;
 }
 
-/* Button on Right */
-.ui.action.input:not([class*="left action"]) > input {
+/* Input when ui Left*/
+.ui[class*="left"].action.input > input {
+  border-top-left-radius: 0 !important;
+  border-bottom-left-radius: 0 !important;
+  border-left-color: transparent !important;
+}
+
+/* Input when ui Right*/
+.ui.action.input:not([class*="left"]) > input {
```

Comment:
Same here, please change to
```css
.ui.action.input:not([class*="left action"]) > input {
```
to not interfere with `<div class="ui right action input left icon">`